src/modals/new-identity-flow/NewIdentityFlow.js

```diff
@@ -0,0 +1,143 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { PromiseState, getPromiseState } from 'react-promiseful';
+
+import { FlowModal, FlowModalStep, Button, TextButton } from '@nomios/web-uikit';
+import GenericStep from './generic-step';
+import { IdentityType, IdentityInfo, Feedback } from './create-identity-steps';
+
+class NewIdentityFlow extends Component {
+    state = {
+        currentStepId: 'generic',
+        currentFlow: undefined,
+        promise: undefined,
+        data: {},
+    };
+
+    render() {
+        const { currentStepId } = this.state;
+
+        return (
+            <FlowModal variant="advanced" step={ currentStepId } { ...this.props }>
+                <FlowModalStep id="generic">
+                    <GenericStep onNextStep={ this.handleChooseNextFlow } />
+                </FlowModalStep>
+                { this.renderFlowSteps() }
+            </FlowModal>
+        );
+    }
+
+    renderFlowSteps() {
+        const { currentFlow } = this.state;
+
+        switch (currentFlow) {
+        case 'create':
+            return this.renderCreateSteps();
+        case 'import':
+            return this.renderImportSteps();
+
+        default: return null;
+        }
+    }
+
+    renderCreateSteps() {
+        const { promise } = this.state;
+
+        const identityFirstName =
+            this.state.data['create-identity-type'] &&
+            this.state.data['create-identity-type'].name &&
+            this.state.data['create-identity-type'].name.split(' ')[0];
+
+        return (
+            <Fragment>
+                <FlowModalStep id="create-identity-type">
+                    <IdentityType
+                        nextStepId="create-identity-info"
+                        onNextStep={ this.handleNextStep } />
+                </FlowModalStep>
+                <FlowModalStep id="create-identity-info">
+                    <IdentityInfo
+                        nextStepId="create-identity-feedback"
+                        onNextStep={ this.handleSubmitForm }
+                        identityFirstName={ identityFirstName } />
+                </FlowModalStep>
+                <FlowModalStep id="create-identity-feedback">
+                    <PromiseState promise={ promise } onSettle={ this.handlePromiseSettle }>
+                        { ({ status }) => (
+                            <Feedback
+                                status={ status }
+                                successActions={ this.renderFeedbackSuccessActions() }
+                                errorActions={ this.renderFeedbackErrorActions() } />
+                        ) }
+                    </PromiseState>
+                </FlowModalStep>
+            </Fragment>
+        );
+    }
+
+    renderFeedbackSuccessActions() {
```

Comment:
I would name this to be more specific to the create identity flow. for example `renderCreateSuccessActions()`

This follows for every function that will have to be replicated for the import identity flow

Reply:
Alternatively, the function can receive an argument saying if it's `import` or `create`.